support/docker/build.sh

```diff
@@ -29,11 +29,136 @@ stderr.printf() {
 	printf "$@" >&2
 }
 
-_nix_docker() {
+
+_nix_container_manager() {
+
+	# Ask if the user wants to proceed as root
+	if (($(id -u) == 0)); 
+	then
+		stderr.printf "\nYou don't need to run this script as root.\n\nLear more on how run docker/podman in rootless mode:\n Docker: https://docs.docker.com/engine/security/rootless/\n Podman: https://github.com/containers/podman/blob/main/docs/tutorials/rootless_tutorial.md\n\n"
+		PS3="Running as root. Do you want to continue?: "
+
+		select yn in "Yes" "No"; do
+			case $yn in
+				Yes) 
+					stderr.printf "\nProceeding as root user\n"
+					break
+					;;
+				No)
+					stderr.printf "\nQuiting the build script...\n"
+					exit 0
+					;;
+			esac
+		done
```

Comment:
Note: This is a case where stopping and asking the user is probably fine? I don't know enough about docker/podman and friends to know if there are drawbacks that warrant stopping running as root.

Reply:
While it would ideally be a good spot to always request permission, most Docker && Podman starter guides are for root setups.
Will change it to always print something visible. But remove the forced interaction by default.